pulp_container/app/registry_api.py

```diff
@@ -473,6 +477,135 @@ def get_queryset(self, *args, **kwargs):
         return (public_repositories | accessible_repositories).distinct()
 
 
+class FlatpakIndexDynamicView(APIView):
+    """
+    Handles requests to the /index/dynamic endpoint
+    """
+
+    authentication_classes = []
+    permission_classes = []
+
+    def recurse_through_manifest_lists(self, tag, manifest, oss, architectures, manifests):
+        mlms = manifest.listed_manifests.through.objects.filter(image_manifest__pk=manifest.pk)
+        if oss:
+            mlms.filter(os__in=oss)
+        if architectures:
+            mlms.filter(architecture__in=architectures)
+        for mlm in mlms:
+            self.recurse_through_manifest_lists(
+                tag, mlm.manifest_list, oss, architectures, manifests
+            )
+        if manifest.media_type in (models.MEDIA_TYPE.MANIFEST_V2, models.MEDIA_TYPE.MANIFEST_OCI):
```

Comment:
Do you think moving this `if` condition to the top of the method is more efficient?

Reply:
Also, I am not quite sure if it is necessary to recursively go through manifests because there should be just one manifest list within the stack.

Reply:
Moved the `if` part to the top in c6605fe350000eaa3c0de35d05f73af25453405d.
I don't know if it is somehow guaranteed that there will only ever be one level of manifest lists, but why not keep the code simple and generic and let it thus automatically cater for multiple levels of lists.